File: 152-maximum-product-subarray/152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        if(nums.length == 0) return 0;
        int n = nums.length;
        
        int ans = Integer.MIN_VALUE;
        for(int i=0;i<n;i++){
            int prod = 1;
            for(int j=i;j<n;j++){
                prod = prod * nums[j];
                ans = Math.max(ans,prod);
            }
        }
        return ans;
    }
}

File: 152-maximum-product-subarray/152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n = nums.size();
        
        int ans = INT_MIN;
        int lprod = 1,rprod = 1;
        for(int i=0;i<n;i++) {
            lprod = lprod * nums[i];
            rprod = rprod * nums[n-i-1];
            ans = max({ans,lprod,rprod});
            
            if(lprod == 0) lprod = 1;
            if(rprod == 0) rprod = 1;
        }
        
        return ans;
    }
};